File: examples/node/hairstyle-pro-async.ts
import { readFileSync } from "node:fs";
import { AILabClient } from "ailabtools";

const client = new AILabClient({ apiKey: process.env.AILAB_API_KEY! });

async function main() {
  const task = await client.portrait.changeHairstyle({
    image: readFileSync("./portrait.jpg"),
    hairStyle: "BuzzCut",
    color: "blonde",
  });

  const taskId = task.task_id || task.data?.task_id;
  if (!taskId) throw new Error("Missing task_id from hairstyle response");

  const result = await client.waitForTask(taskId, {
    intervalMs: 5000,
    timeoutMs: 300000,
  });

  console.log(result.data);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: examples/python/hairstyle_pro_async.py
import asyncio
import os
from ailabtools import AILabClient


async def main():
    client = AILabClient(api_key=os.environ["AILAB_API_KEY"])
    with open("./portrait.jpg", "rb") as image:
        task = await client.portrait.change_hairstyle({
            "image": image,
            "hairStyle": "BuzzCut",
            "color": "blonde",
        })

    task_id = task.get("task_id") or task.get("data", {}).get("task_id")
    if not task_id:
        raise RuntimeError("Missing task_id from hairstyle response")

    result = await client.wait_for_task(task_id, interval=5, timeout=300)
    print(result.get("data"))
    await client.aclose()


asyncio.run(main())
